test/utils.py

```diff
@@ -0,0 +1,51 @@
+from collections.abc import Sequence
+from dataclasses import dataclass
+
+import numpy as np
+from gymnasium import spaces
+
+from tianshou.data.collector import CollectStats
+
+
+@dataclass
+class SpaceInfo:
+    action_shape: int | Sequence[int]
+    state_shape: int | Sequence[int]
+    action_dim: int
+    state_dim: int
```

Comment:
state -> obs

Isn't action_dim and state_dim tied to the action_shape and state_shape? I don't fully understand why we need both.

If we do need, it probably should be a property instead of a field that's being passed.

Reply:
> state -> obs

I agree, and it is no hassle to change it here, but should we also change then the names of the args in the test files, where they are named `args.state_shape` (that might be too much of refactoring in this PR, I think).

Reply:
Yes, opening a followup issue. There might be other places in the code where state_shape is used